src/libcommonserver/io/iohelper_win.cpp

```diff
@@ -442,39 +442,26 @@ static bool getRightsWindowsApi(const SyncPath &path, bool &read, bool &write, b
 
     // Get rights for trustee
     ACCESS_MASK rights = 0;
-    EXPLICIT_ACCESS *explicitAccessList = nullptr;
-    auto explicitAccessCount = std::make_unique<ULONG>(0);
-    result = GetExplicitEntriesFromAcl(pfileACL, explicitAccessCount.get(), &explicitAccessList);
+    result = GetEffectiveRightsFromAcl(pfileACL, &Utility::_trustee, &rights);
     ioError = dWordError2ioError(result);
-    if (ioError != IoErrorSuccess) {
-        LOGW_WARN(logger, L"GetExplicitEntriesFromAcl failed: path='" << Utility::formatSyncPath(path) << L"',DWORD err='"
-                                                                      << result << L"'");
-        LocalFree(explicitAccessList);
+    if (result == ERROR_INVALID_ACL) { // The GetEffectiveRightsFromAcl function fails and returns ERROR_INVALID_ACL if the specified ACL contains an inherited access-denied ACE.
```

Comment:
Could you explain why it is an acceptable choice? It would be nice if you could add a link to the Windows documentation where this behavior is described.

Reply:
https://github.com/Infomaniak/desktop-kDrive/pull/115